### ISSUE TITLE
bug: Google Play Services error reporting is undescriptive

### ISSUE THREAD

**Issue by erikwalle** [robingenz/capacitor-firebase-authentication#85]
**Plugin version:**
v0.3.11

**Platform(s):**
Android

**Current behavior:**
FirebaseAuthentication.signInWithPlayGames() generates undescriptive error messages of the form: "Error: 4: " (Where the number can be 4 or 8 from what I've seen so far). Anecdotally, "Error: 4: " seems to be thrown when login is attempted without network connectivity.
Checking logs in Android studio, there is no additional reporting that helps narrow down the problem.

**Expected behavior:**
Error messages that describe the problem.

**Steps to reproduce:**
Set phone to airplane mode and call "FirebaseAuthentication.signInWithPlayGames()", observe error logs.

**Comment by robingenz** [comment]
Hi @erikwalle,
thank you for reporting this.
I found the following descriptions in the [Google Play services documentation](https://developers.google.com/android/reference/com/google/android/gms/common/api/CommonStatusCodes.html):
- **4**:
  > The client attempted to connect to the service but the user is not signed in. The client may choose to continue without using the API. Alternately, if Status.hasResolution() returns true the client may call Status.startResolutionForResult(Activity, int) to prompt the user to sign in. After the sign in activity returns with Activity.RESULT_OK further attempts should succeed.
  
  This description fits with your assertion that this always occurs when there is no internet connection, as the user cannot sign in without a connection.
  
- **8 (INTERNAL_ERROR)**:
  > An internal error occurred. Retrying should resolve the problem.
  
  I think we can ignore this error code at least for now.


Basically, of course, I can't do anything about these error messages except describe them better and assign them a unique code.
For this I have created the following issue: #86 
Other ideas are welcome.
Since this is a feature from my point of view, I have created a new issue and am closing this one.

**Comment by erikwalle** [comment]
@robingenz Any ETA for v1.0? Honesty appreciated, thanks.

**Comment by robingenz** [comment]
My honest answer: I cannot say this at the moment, because there are still a lot of features planned.
I will prioritize issue #86.

**Comment by robingenz** [comment]
I guess about 3 months. 🤔 

Apart from issue #86, are you waiting for anything specific?
I may be able to adjust the priorities.

**Comment by erikwalle** [comment]
I want to gauge how production-ready this plugin is. There's not a lot at stake since I am still in beta, and I'm 90% sure I want to go with Firebase login instead of raw Play Services. So I will probably proceed with the rollout anyway.

My worst case scenario is if this plugin has a fatal flaw (since I don't see many alternatives). I can't think of any other features I need in terms of auth at this time, but legible error reporting feels like an important one. Knowing the numbers that are rendered (a la "Error: #: ") correspond to GPS error codes is helpful.

**Comment by robingenz** [comment]
Okay I understand. Yes, issue #86 is definitely very important. I'll do my best to implement it soon.
Other than that, all I can say is that I don't know of any bugs at this time. These have the highest priority for me (also since I use the plugin myself productively).
